### PR TITLE
Restyle register

### PR DESCRIPTION
**Pull request by curioussavage**
work in progress...

### CODE REVIEW

**Comment by curioussavage** [comment]
:eyeglasses: @ajacksified @nramadas

**Comment by curioussavage** [comment]
Would definitely recommend going through by commit. When I was working on the token refresh client side I felt like the giant client file made it hard to follow along with so I broke that out too. Mostly shuffling code around, with a few changes.

Another change is the old modal component is stateless. 

Its on staging and needs a little polishing still. Just wanted to get reviews started now since the diff is huge.

**Comment by ajacksified** [comment]
* when I first log in, when I click to upvote something, it takes me to register
* when I hit 'enter' / submit from the password field, it runs the "view password" action instead of submitting the form

:nail_care:

File: assets/less/components/LoginPage.less [hunk 1]
```diff
@@ -1,4 +1,66 @@
...
+    margin-right: -5px;
+  }
+}
+
+.login-register-link {
+  text-align: center;
+  padding-top: 5px;
+}
+
+.login-form {
```
Comment (ajacksified) [inline assets/less/components/LoginPage.less]: kill this class if it's empty

**Comment by ajacksified** [comment]
Erg, sorry, added comments on commits rather than on the PR. Be nice if github automatically resolved them.

**Comment by curioussavage** [comment]
img_0797
img_0796
img_0795

**Comment by curioussavage** [comment]
darn, I rebased the big commit with the client stuff into the login stuff. going to try to fix that

**Comment by curioussavage** [comment]
issue with voting and other actions after login should be resolved.

**Comment by curioussavage** [comment]
:haircut:  I am still converting the other less files to the new style but the major issues were resolved

**Comment by curioussavage** [comment]
CSS is looking much better now.

**Comment by curioussavage** [comment]
paging @librarianavenger

**Comment by curioussavage** [comment]
:eyeglasses: @schwers could you look at that last commit

File: assets/js/lib.es6.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  const totalScroll = top + el.offsetHeight;
+  if (totalScroll === el.scrollHeight) {
+    el.scrollTop = top - 1;
+    return false;
+  }
+
+  return true;
+}
+
+export const findLinkParent = function findLinkParent(el) {
```
Comment (nramadas) [inline assets/js/lib.es6.js]: out of curiosity, why declare these as named function expressions? wouldn't `export function findLinkParent(el) {...}` work just as well?

File: src/views/components/forgotpassword.jsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      this.setState({
+        error: ERRORS[e.name] ||
+               ERRORS[e.status] ||
+               ERRORS.DEFAULT,
+      });
+    }
+  }
+
+  async makeRequest(uri, postData) {
+    return await new Promise(async function(resolve, reject) {
```
Comment (nramadas) [inline src/views/components/forgotpassword.jsx]: async functions already return promises, so I don't think you need to wrap it in a promise. you could turn it into an IIFE instead:

```
return await (async function() {...})();

File: assets/js/lib.es6.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
+import superagent from 'superagent';
```
Comment (schwers) [inline assets/js/lib.es6.js]: :no_bell: Would you be opposed to renaming this to something like `clientUtils` or `clientLib` to disambiguate it from `lib/` ?

File: src/views/components/forgotpassword.jsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      } catch (e) {
+        if (e.timeout) {
+          e.name = 504;
+        }
+        return reject(e);
+      }
+    });
+  }
+
+  updateName(e, val) {
```
Comment (nramadas) [inline src/views/components/forgotpassword.jsx]: it looks like `updateName` is either being fired as a callback or by `clear`. Since `val` is always either undefined or `''`, I think we can drop it and just say:
```
const name = e ? e.target.value : '';
```

File: src/views/components/forgotpassword.jsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+      </button>
+    );
+  }
+
+  clear() {
+    this.updateName(null, '');
+  }
+
+  render() {
+    const name = this.state.name || '';
```
Comment (nramadas) [inline src/views/components/forgotpassword.jsx]: why not default this in the constructor?

File: src/views/components/snooiconheader.jsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+function SnooIconHeader(props) {
```
Comment (nramadas) [inline src/views/components/snooiconheader.jsx]: can we add some type checks? I know there are only two props, but for consistency, we should always add these checks.

File: src/views/pages/login.jsx [hunk 8]
```diff
@@ -1,15 +1,69 @@
...
+};
+
+const placeholderText = {
+  register: {
+    username: 'Choose a unique username',
+    password: 'Choose a unique password',
+    email: 'Your email',
+  },
+  login: {
+    username: 'username',
```
Comment (nramadas) [inline src/views/pages/login.jsx]: nitpick, but should these be capitalized to match the copy in the register form?

File: src/views/pages/login.jsx [hunk 9]
```diff
@@ -1,15 +1,69 @@
...
+    { 'Privacy Policy ' }
+  </a>
+);
+const content = (
+  <a
+    href='/help/contentpolicy/'
+    className='text-link'
+    target='_blank'
+  >
+    { 'Content Policy' }
```
Comment (nramadas) [inline src/views/pages/login.jsx]: Couldn't you drop the curly brackets?

File: src/views/pages/login.jsx [hunk 10]
```diff
@@ -1,15 +1,69 @@
...
-  500: 'There was a problem with the server',
-  400: 'Wrong password',
-  default: 'An error occured.',
+  504: 'Sorry, it took too long for the server to respond',
+  500: 'Sorry, something has gone wrong with the server',
+  WRONG_PASSWORD: 'Sorry, that’s not the right password',
+  default: 'Sorry, Something has gone wrong and we\'re not sure what',
+};
+
+const PasswordFieldTypes = {
```
Comment (nramadas) [inline src/views/pages/login.jsx]: :no_bell: just a personal preference of mine, I like naming true constants with all uppercase, like you did for `ERROR_MESSAGES`. So,
```
const PASSWORD_FIELD_TYPES = {
    PASSWORD: 'password',
    TEXT: 'text',
};
```

File: src/views/components/forgotpassword.jsx (hunk 3 above)
Comment (schwers) [inline src/views/components/forgotpassword.jsx]: @nramadas Does superagent or babel do the promise wrapping now automatically? There's a plugin available to add real promises with .then and in most of the code we wrap it in Promises ourselves. If we can just await it across the board -- would be a nice clean up patch

Comment (nramadas) [inline src/views/components/forgotpassword.jsx]: @schwers I believe babel (through the regenerator runtime) adds the promises in for us: https://github.com/facebook/regenerator/blob/master/runtime.js#L136

The annoying part is that this isn't revealed in the transpiled code. I believe the promise is generated at runtime.

Comment (ajacksified) [inline src/views/components/forgotpassword.jsx]: Or, we switch to fetch \o/

File: src/views/pages/login.jsx (hunk 10 above)
Comment (ajacksified) [inline src/views/pages/login.jsx]: Yeah, I thought it was in the style guide (but I guess not), but file-global constants should be uppercased.

Comment (curioussavage) [inline src/views/pages/login.jsx]: yep, good catch. something I forgot to clean up

File: src/views/pages/login.jsx (hunk 9 above)
Comment (curioussavage) [inline src/views/pages/login.jsx]: not without losing the spaces. I would prefer this to adding &nbsp;

File: src/views/components/forgotpassword.jsx (hunk 6 above)
Comment (curioussavage) [inline src/views/components/forgotpassword.jsx]: good catch, another thing that wasn't cleaned up.

File: assets/js/lib.es6.js (hunk 2 above)
Comment (curioussavage) [inline assets/js/lib.es6.js]: Thanks for pointing this one out. For some reason I had in my head the idea that you had to assign something to a variable to use named exports

File: src/views/components/forgotpassword.jsx (hunk 3 above)
Comment (curioussavage) [inline src/views/components/forgotpassword.jsx]: fetch would be better. If we rely on babel doing that at runtime then its more work if we wan't to start running without babel on the server.

**Comment by curioussavage** [comment]
:haircut: all

File: src/views/pages/login.jsx [hunk 11]
```diff
@@ -17,82 +72,379 @@ class LoginPage extends BasePage {
...
+    //   // set global error
+    // }
+
+    const message = e.message || ERROR_MESSAGES[e.error];
+    this.setState({ error: newError, errorMessage: message });
+  }
+
+  makeRequest(endpoint, data) {
+    return new Promise((resolve, reject) => {
+      superagent
```
Comment (nramadas) [inline src/views/pages/login.jsx]: Does superagent return promises? If so, we can avoid wrapping the api call here in another promise.

Comment (ajacksified) [inline src/views/pages/login.jsx]: It does not.

Comment (curioussavage) [inline src/views/pages/login.jsx]: although we could start just wrapping this stuff in async calls since apparently regenerator will wrap it with a generator.

Comment (schwers) [inline src/views/pages/login.jsx]: We should be careful with testing that. I'm curious if it's smart enough to work with superagent specifically. I'm guessing they assume that there's a `.then()` method of what you're awaiting and superagent has a [faux-promise-then method](https://github.com/visionmedia/superagent/blob/master/lib/request-base.js#L56). I'm of the opinion that it's okay to move that into a separate 'cleanup-superagent-calls' patch OR wait till the new `fetch` work happens

**Comment by curioussavage** [comment]
:eyeglasses: @spez because this is a visual change.

this is live on https://www.cloudflarestagingformobilereddit.com/ right now

File: src/views/pages/login.jsx (hunk 11 above)
Comment (ajacksified) [inline src/views/pages/login.jsx]: :+1: to waiting

**Comment by spez** [comment]
:fish:

**Comment by curioussavage** [comment]
:haircut:

File: assets/less/components/LoginPage.less [hunk 12]
```diff
@@ -1,4 +1,71 @@
...
+  &__wrapper {
+    background-color: @off-white;
+
+    .row {
+      margin-right: -5px;
+    }
+  }
+
+  &__register-link {
+    color: @alien-blue-text;
```
Comment (ajacksified) [inline assets/less/components/LoginPage.less]: Are all links supposed to be this color by default now? Can we just apply this to the base `a` color?

File: assets/less/components/LoginPage.less [hunk 13]
```diff
@@ -1,4 +1,71 @@
...
+    text-align: center;
+    padding-top: 5px;
+  }
+
+  &__forgot-link {
+    color: @alien-blue-text;
+    margin-top: 5px;
+  }
+
+  &__submit-btn {
```
Comment (ajacksified) [inline assets/less/components/LoginPage.less]: The whole squarebutton class seems needlessly overcomplex, but I guess now isn't the time to fix it. But... wut. :no_bell:

File: assets/less/components/LoginPage.less [hunk 14]
```diff
@@ -1,4 +1,71 @@
...
+
+    [class^="icon-"] {
+      font-size: 24px;
+    }
+
+    .icon-x {
+      color: @red;
+    }
+
+    &.blue {
```
Comment (ajacksified) [inline assets/less/components/LoginPage.less]: `.blue [class^=icon]` is sufficient - no need to nest with `&` here. Move this into `text.less`.

File: assets/less/components/Modal.less [hunk 15]
```diff
@@ -1,8 +1,19 @@
...
 	margin: 130px auto 30px;
-}
\ No newline at end of file
+}
+
+.modal-content {
+  border-radius: 0;
+}
+
+.Modal__close {
```
Comment (ajacksified) [inline assets/less/components/Modal.less]: the mixing of cases is literally making me cry :no_bell:

File: assets/less/components/minimalinput.less [hunk 16]
```diff
@@ -0,0 +1,42 @@
+.inputMinimal {
```
Comment (ajacksified) [inline assets/less/components/minimalinput.less]: the file is `minimalInput`, the class is `inputMinimal`. I'd suggest changing the filename

File: assets/less/components/snooiconheader.less [hunk 17]
```diff
@@ -0,0 +1,24 @@
+.SnooIconHeader-wrapper {
```
Comment (ajacksified) [inline assets/less/components/snooiconheader.less]: If we're moving towards camelcase for new css, can we camelcase all of this without breaking anything existing?

File: src/server/index.jsx [hunk 18]
```diff
@@ -28,7 +28,7 @@ import setLoggedOutCookies from '../lib/loid';
 
 import Config from '../config';
 
-const ignoreCSRF = ['/timings', '/error'];
+const ignoreCSRF = ['/timings', '/error', '/login', '/register'];
```
Comment (ajacksified) [inline src/server/index.jsx]: shouldn't have to ignore it now that we're not cdn-cached.

Comment (kjoconnor) [inline src/server/index.jsx]: I wouldn't assume we'll always be running uncached.

Comment (ajacksified) [inline src/server/index.jsx]: If we're implementing logged-out oauth, we're going to have to.

**Comment by ajacksified** [comment]
Three minor things; :nail_care:

File: assets/less/components/LoginPage.less (hunk 12 above)
Comment (curioussavage) [inline assets/less/components/LoginPage.less]: erica and I talked about this, Ill check again and if that was the case will definitely do that.

File: assets/less/components/LoginPage.less (hunk 14 above)
Comment (schwers) [inline assets/less/components/LoginPage.less]: The color modifiers in `icon-font.less` should take care of this for you. You should be able to remove this rule altogether and add the class `blue` to the icon itself and get the same effect.

File: assets/less/components/snooiconheader.less (hunk 17 above)
Comment (curioussavage) [inline assets/less/components/snooiconheader.less]: did you mean camelcase for the filename or were you meaning to say use __ for bem syntax?

Comment (ajacksified) [inline assets/less/components/snooiconheader.less]: Yes, and also snooIconHeader

Comment (ajacksified) [inline assets/less/components/snooiconheader.less]: err, maybe we're not. I'm not sure. I thought we were moving to camelcase, but @nramadas just added a bunch of stuff with capitals in https://github.com/reddit/reddit-mobile/pull/506/files#diff-7ad847eaa65fdf0831fdc66da8abb916R1

File: assets/less/components/minimalinput.less (hunk 16 above)
Comment (ajacksified) [inline assets/less/components/minimalinput.less]: and we should figure out capitalization

**Comment by curioussavage** [comment]
:haircut: addressed issues, also tweaked some css a little to increase hit area on all links/ close buttons

**Comment by ajacksified** [comment]
:fish:
